src/main/java/teammates/common/util/TimeKeeper.java

```diff
@@ -0,0 +1,29 @@
+package teammates.common.util;
+
+import java.util.Date;
+
+import com.google.apphosting.api.ApiProxy;
+import com.google.apphosting.api.DeadlineExceededException;
+
+public class TimeKeeper {
+    private static final long REMAINING_TIME_THRESHOLD = 5000;
```

Comment:
This can be taken as a parameter in the constructor? This is because the value may differ based on the place where we use it.